Adafruit_FXAS21002C.cpp

```diff
@@ -263,3 +281,97 @@ void Adafruit_FXAS21002C::standby(boolean standby) {
     active_bit.write(0x03);
   }
 }
+
+/**************************************************************************/
+/*!
+    @brief  Configures the device with certain output data rate(ODR)
+            Supports ODRs: 8000Hz, 400Hz, 200Hz, 100Hz, 50Hz, 25Hz, 12.5Hz
+    @param   ODR : the output data rate to be set to the gyroscope
+*/
+/**************************************************************************/
+void Adafruit_FXAS21002C::setODR(float ODR) {
+  Adafruit_BusIO_Register CTRL_REG1(i2c_dev, GYRO_REGISTER_CTRL_REG1);
+  Adafruit_BusIO_RegisterBits datarate_bits(&CTRL_REG1, 3, 2);
+
+  /* CTRL_REG1 should only be set in Standby or Ready mode. First enter Standby
+   * mode */
+  standby(true);
+
+  if (ODR == GYRO_ODR_800HZ) {
+    datarate_bits.write(0b000);
+    _ODR = ODR;
+  } else if (ODR == GYRO_ODR_400HZ) {
+    datarate_bits.write(0b001);
+    _ODR = ODR;
+  } else if (ODR == GYRO_ODR_200HZ) {
+    datarate_bits.write(0b010);
+    _ODR = ODR;
+  }
+
+  else if (ODR == GYRO_ODR_100HZ) {
+    datarate_bits.write(0b011);
+    _ODR = ODR;
+  }
+
+  else if (ODR == GYRO_ODR_50HZ) {
+    datarate_bits.write(0b100);
+    _ODR = ODR;
+  }
+
+  else if (ODR == GYRO_ODR_25HZ) {
+    datarate_bits.write(0b101);
+    _ODR = ODR;
+  }
+
+  else if (ODR == GYRO_ODR_12_5HZ) {
+    datarate_bits.write(0b110);
+    _ODR = ODR;
+  }
+
+  standby(false);
+}
+
+/**************************************************************************/
+/*!
+    @brief  Obtain the current output data rate(ODR) from the gyroscope's
+   register
+    @return The Output Data Rate(ODR) in Hz
+*/
+/**************************************************************************/
+float Adafruit_FXAS21002C::getODR() {
+  Adafruit_BusIO_Register CTRL_REG1(i2c_dev, GYRO_REGISTER_CTRL_REG1);
+  Adafruit_BusIO_RegisterBits datarate_bits(&CTRL_REG1, 3, 2);
+
+  /* Read the current value in the Output Data Rate(ODR) bits and return the
+   * corresponding ODR in Hz */
+  switch (datarate_bits.read()) {
+  case 0b000:
+    _ODR = GYRO_ODR_800HZ;
+    break;
+  case 0b001:
+    _ODR = GYRO_ODR_400HZ;
+    break;
+  case 0b010:
+    _ODR = GYRO_ODR_200HZ;
+    break;
+  case 0b011:
+    _ODR = GYRO_ODR_100HZ;
+    break;
+  case 0b100:
+    _ODR = GYRO_ODR_50HZ;
+    break;
+  case 0b101:
+    _ODR = GYRO_ODR_25HZ;
+    break;
+  case 0b110:
+    _ODR = GYRO_ODR_12_5HZ;
+    break;
+  case 0b111:
+    _ODR = GYRO_ODR_12_5HZ;
+  /* The default case will never occur. It is here to accommodate different
+   * compilers */
+  default:
+    break;
+  }
+  return _ODR;
```

Comment:
Yep. Sorry, wasn't clear - can then delete all the other lines that do the register read and decoding. So this function will just be a one liner - the return statement. Since `_ODR` gets set in `setODR()`, no need to read the register.

Reply:
Yes I will update this function. Thanks!